app/scripts/transaction-manager.js

```diff
@@ -240,17 +256,26 @@ module.exports = class TransactionManager extends EventEmitter {
   //  checks if a signed tx is in a block and
   // if included sets the tx status as 'confirmed'
   checkForTxInBlock () {
-    var signedTxList = this.getFilteredTxList({status: 'signed', err: undefined})
+    var signedTxList = this.getFilteredTxList({status: 'signed'})
     if (!signedTxList.length) return
     signedTxList.forEach((tx) => {
       var txHash = tx.hash
       var txId = tx.id
-      if (!txHash) return
+      if (!txHash) {
+        tx.err = {
+          errCode: 'No hash was provided',
+          message: 'Tx could possibly have not been submitted or an error accrued during signing',
```

Comment:
This case should never occur.  Instead, we should make sure we never set a tx as `signed` when it isn't signed.

Sounds like we can avoid this by submitting the tx once we've signed it ourselves, instead of passing it to the provider-engine for submission, since this signing might be happening after provider-engine has been deallocated.